src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -344,6 +431,8 @@ static void report_broker_config(
       "listeners",
       config::node().kafka_api,
       include_synonyms,
+      maybe_make_documentation(
```

Comment:
do you need `maybe_make_documentation`? You can just always pass the documentation and the bool and then right before you create the response struct you just check the boolean. No need for optionals

Reply:
I used the optional in response to this review comment from a week ago
https://github.com/redpanda-data/redpanda/pull/7543#discussion_r1038456729

And I happen to agree with this style because:
 1. We do not need `include_documentation` as a parameter. The signatures to `add_topic_config` and `add_broker_config` are already large and more params make things harder-to-read.
 2. The documentation object within `describe_configs_resource_result` is an optional. So by using this style I can just directly assign the object without any other checks.

Reply:
Ah true, i was confused with the other bool `include_synonyms`